src/compiler/binder.ts

```diff
@@ -118,11 +118,7 @@ module ts {
         }
 
         function declareSymbol(symbols: SymbolTable, parent: Symbol, node: Declaration, includes: SymbolFlags, excludes: SymbolFlags): Symbol {
-            // Nodes with computed property names will not get symbols, because the type checker
-            // does not make properties for them.
-            if (hasComputedNameButNotSymbol(node)) {
-                return undefined;
-            }
+            Debug.assert(!hasComputedNameButNotSymbol(node));

```

Comment:
So ensure that either it does not have a computed name OR if it does, it has a symbol? I'm confused.